llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -15482,6 +15499,24 @@ SDValue DAGCombiner::visitFREEZE(SDNode *N) {
     }
   }
 
+  // Special case handling for ShuffleVectorSDNode nodes.
```

Comment:
Why isn't it enough to add VECTOR_SHUFFLE to AllowMultipleMaybePoisonOperands?

Reply:
I'll take another look - this patch started out a little different and I think we can reduce the diff even further.